cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -348,6 +348,10 @@ protected void startUp() throws Exception {
         Transactions.invalidateQuietly(txClient, tx);
         throw t;
       }
+    } catch (LinkageError e) {
+      // Need to wrap LinkageError. Otherwise, listeners of this Guava Service may not be called if the initialization
+      // of the user program is missing dependencies (CDAP-2543)
+      throw new Exception(e.getMessage(), e);
```

Comment:
The exception message is then not the same, so I figured it would be clear this way.
See https://github.com/caskdata/cdap/pull/7861#discussion_r99976665 for reference on a discussion I had with @ane .